File: GCDUserInterface/ChangeDetection/frmCoherenceProperties.vb
Namespace ChangeDetection

    Public Class frmCoherenceProperties

        Private m_nFilterSize As Integer

        Public Property FilterSize As Integer
            Get
                Return m_nFilterSize
            End Get
            Set(ByVal value As Integer)
                m_nFilterSize = value
            End Set
        End Property

        Public Property PercentLess As Integer
            Get
                Return numLess.Value
            End Get
            Set(ByVal value As Integer)
                numLess.Value = value
            End Set
        End Property

        Public Property PercentGreater As Integer
            Get
                Return numGreater.Value
            End Get
            Set(ByVal value As Integer)
                numGreater.Value = value
            End Set
        End Property

        Private Sub CoherencePropertiesForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

            Dim sFilterSizeText As String = m_nFilterSize.ToString & " x " & m_nFilterSize.ToString
            cboFilterSize.SelectedItem = sFilterSizeText
        End Sub

        Private Sub cboFilterSize_SelectedIndexChanged(sender As Object, e As System.EventArgs) Handles cboFilterSize.SelectedIndexChanged

            Dim i As Integer = cboFilterSize.Text.IndexOf(" ")
            If i > 0 Then
                m_nFilterSize = cboFilterSize.Text.Substring(0, i)
            End If
        End Sub

        Public Sub New()

            ' This call is required by the designer.
            InitializeComponent()

            ' Default filter size.
            m_nFilterSize = 5
        End Sub

        Private Sub btnHelp_Click(sender As System.Object, e As System.EventArgs) Handles btnHelp.Click
            'Process.Start(My.Resources.HelpBaseURL & "")
        End Sub
    End Class

End Namespace

File: GCDUserInterface.ConvertedToC#/ChangeDetection/frmCoherenceProperties.cs
using GCDCore.Project;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
namespace GCDUserInterface.ChangeDetection
{

	public partial class frmCoherenceProperties
	{


		private int m_nFilterSize;
		public int FilterSize {
			get { return m_nFilterSize; }
			set { m_nFilterSize = value; }
		}

		public int PercentLess {
			get { return numLess.Value; }
			set { numLess.Value = value; }
		}

		public int PercentGreater {
			get { return numGreater.Value; }
			set { numGreater.Value = value; }
		}


		private void CoherencePropertiesForm_Load(System.Object sender, System.EventArgs e)
		{
			string sFilterSizeText = m_nFilterSize.ToString() + " x " + m_nFilterSize.ToString();
			cboFilterSize.SelectedItem = sFilterSizeText;
		}


		private void cboFilterSize_SelectedIndexChanged(object sender, System.EventArgs e)
		{
			int i = cboFilterSize.Text.IndexOf(" ");
			if (i > 0) {
				m_nFilterSize = cboFilterSize.Text.Substring(0, i);
			}
		}


		public frmCoherenceProperties()
		{
			Load += CoherencePropertiesForm_Load;
			// This call is required by the designer.
			InitializeComponent();

			// Default filter size.
			m_nFilterSize = 5;
		}

		private void btnHelp_Click(System.Object sender, System.EventArgs e)
		{
			//Process.Start(My.Resources.HelpBaseURL & "")
		}
	}

}
